lib/buildpackutil.py

```diff
@@ -122,9 +124,11 @@ def download_and_unpack(url, destination, cache_dir='/tmp/downloads'):
             ['dpkg-deb', '-x', cached_location, destination]
         )
     elif file_name.endswith('.tar.gz') or file_name.endswith('.tgz'):
-        subprocess.check_call(
-            ['tar', 'xf', cached_location, '-C', destination]
-        )
+        unpack_cmd = ['tar', 'xf', cached_location, '-C', destination]
+        if 'mono' in file_name:
```

Comment:
it's better to make this a more strict condition: `filename.startswith("mono-")` because the current condition is too loose.